app/uk/gov/hmrc/gform/dms/DmsSubmissionController.scala

```diff
@@ -44,36 +49,68 @@ class DmsSubmissionController(
     withJsonBody[DmsHtmlSubmission] { dmsHtmlSubmission =>
       val decodedHtml = decode(dmsHtmlSubmission.html)
       val metadata = dmsHtmlSubmission.metadata
-      val formTemplateId = FormTemplateId(metadata.dmsFormId)
 
-      for {
-        envId <- fileUpload.createEnvelope(formTemplateId, LocalDateTime.now(clock).plusDays(config.formExpiryDays))
-        pdf   <- pdfGenerator.generatePDF(decodedHtml)
-        pdfDoc = documentLoader(pdf)
-        pdfSummary = PdfSummary(pdfDoc.getNumberOfPages.toLong, pdf)
-        _ = pdfDoc.close()
-        submissionRef = SubmissionRef(envId)
-        dmsMetadata = DmsMetaData(formTemplateId, metadata.customerId)
-        submission = Submission(
-          FormId(metadata.dmsFormId),
-          LocalDateTime.now(clock).plusDays(config.formExpiryDays),
-          submissionRef,
-          envId,
-          0,
-          dmsMetadata)
-        // TODO controller should allow data XML submission
-        summaries = PdfAndXmlSummaries(pdfSummary)
-        dmsSubmission = DmsSubmission(
-          metadata.dmsFormId,
-          TextExpression(Constant(metadata.customerId)),
-          metadata.classificationType,
-          metadata.businessArea)
-        _ <- fileUpload.submitEnvelope(submission, summaries, dmsSubmission, 0)
-      } yield {
-        NoContent
+      pdfGenerator.generatePDF(decodedHtml).flatMap { byteArray =>
+        submit(byteArray, metadata)
       }
     }
   }
 
+  /**
+    * This endpoint is not used by gform but its here for other services to leverage our backend integration with DMS via FUaaS
+    * Its currently used by Overseas Agents team
+    */
+  def submitPdfToDms: Action[MultipartFormData[Files.TemporaryFile]] = Action.async(parse.multipartFormData) {
+    implicit request =>
+      def validContentType(filePart: MultipartFormData.FilePart[TemporaryFile]) =
+        filePart.contentType.map(_.toLowerCase).contains("application/pdf")
+
+      request.body.files.headOption match {
+        case Some(file) if validContentType(file) =>
+          val dataParts = request.body.dataParts.mapValues(_.mkString(""))
+          Json.toJson(dataParts).validate[DmsMetadata] match {
+            case JsSuccess(metadata, _) =>
+              val file = request.body.files.head.ref.file.getAbsolutePath
```

Comment:
use `file` from line 69 instead of `request.body.files.head`

Reply:
ah sorry, missed it, 
pushed the fix